.circleci/config.yml

```diff
@@ -0,0 +1,193 @@
+            --enable-option-checking=fatal \
+            --prefix=/usr \
+            --enable-phpdbg \
+            --enable-fpm \
+            --enable-opcache \
+            --with-pdo-mysql=mysqlnd \
+            --with-mysqli=mysqlnd \
+            --with-pgsql \
+            --with-pdo-pgsql \
+            --with-pdo-sqlite \
+            --enable-intl \
+            --without-pear \
+            --enable-gd \
+            --with-jpeg \
+            --with-webp \
+            --with-freetype \
+            --with-xpm \
+            --enable-exif \
+            --with-zip \
+            --with-zlib \
+            --with-zlib-dir=/usr \
+            --enable-soap \
+            --enable-xmlreader \
+            --with-xsl \
+            --with-tidy \
+            --enable-sysvsem \
+            --enable-sysvshm \
+            --enable-shmop \
+            --enable-pcntl \
+            --with-readline \
+            --enable-mbstring \
+            --with-curl \
+            --with-gettext \
+            --enable-sockets \
+            --with-bz2 \
+            --with-openssl \
+            --with-gmp \
+            --enable-bcmath \
+            --enable-calendar \
+            --enable-ftp \
+            --with-pspell=/usr \
+            --with-enchant=/usr \
+            --with-kerberos \
+            --enable-sysvmsg \
+            --with-ffi \
+            --enable-zend-test \
+            --enable-dl-test=shared \
+            --with-ldap \
+            --with-ldap-sasl \
+            --with-password-argon2 \
+            --with-mhash \
+            --with-sodium \
+            --enable-dba \
+            --with-cdb \
+            --enable-flatfile \
+            --enable-inifile \
+            --with-tcadb \
+            --with-lmdb \
+            --with-qdbm \
+            --with-snmp \
+            `#--with-unixODBC` \
+            --with-imap \
+            --with-kerberos \
+            --with-imap-ssl \
+            `#--with-pdo-odbc=unixODBC,/usr` \
+            `#--with-pdo-oci=shared,instantclient,/opt/oracle/instantclient` \
+            `#--with-oci8=shared,instantclient,/opt/oracle/instantclient` \
+            --with-config-file-path=/etc \
+            --with-config-file-scan-dir=/etc/php.d \
+            --with-pdo-firebird \
+            `#--with-pdo-dblib` \
+            --disable-phpdbg \
+            `#--enable-werror`
+      - run: make -j$(/usr/bin/nproc) > /dev/null
+      - run: sudo make install
+      - run: sudo mkdir -p /etc/php.d
+      - run: sudo chmod 777 /etc/php.d
+      - run: echo opcache.enable_cli=1 > /etc/php.d/opcache.ini
+      - run: echo opcache.protect_memory=1 >> /etc/php.d/opcache.ini
+      # Specify opcache.preload_user as we're running as root.
+      # - run: echo opcache.preload_user=root >> /etc/php.d/opcache.ini
+      - run: |
+          echo $LANGUAGE
+          echo $LANG
+          echo $LC_MESSAGES
+          echo $LC_ALL
+      - run: |
+          sapi/cli/php run-tests.php \
```

Comment:
it might make sense to use a different job for this. We could then use `parallelism` to split it which should be much faster.

Reply:
Do we need it? I don't think this build will be the limiting factor. I suppose we could split by extensions. The example in the docs splits by file, I wonder how well that works with thousands of tests? (Especially where 1 file is 1 test)

Reply:
it's about parallelism of the whole job which would be done only for test run. But think that's something that we could add later as it might consume more credits potentially - it saves time though. Let's first do it without that.

Reply:
Yeah, what I meant is that the ARM build isn't the longest build, so it's unlikely somebody is waiting for it. I think this would be great if we moved other builds to CircleCI. But while GitHub actions sucks I don't think the investment would be worth it, especially given that it's not free.

Reply:
What I meant is that parallelism is on job level so we wouldn't need to to do the build multiple times but just once, persist it to workspace and then use job with parallelism (multiple containers) to run tests. We would probably need to add some logic for splitting. Although not sure if it would be that much better than usual splitting in single container - it might just allow a bit better separation as it is on a container level.

Reply:
@bukka Ah ok, yeah that's fair. Most of our builds have differing configurations (with/without zts, release/debug, asan, architectures, etc). Some of the nightly builds do have redundant build steps, but it's not too many.